process-manager.js

```diff
@@ -42,19 +42,30 @@ class ProcessWrapper extends EventEmitter {
 	}
 }
 
+// execFile - the path to the file to spawn the child process(es) from
+// maxProcesses - the maximum number of child processes to spawn
+// isChatBased - the process manager handles some chat functionality
 class ProcessManager {
 	constructor(options) {
-		if (!options) options = {};
+		if (!('execFile' in options)) {
+			Monitor.debug(
+				"No execFile property was missing form the options object to be " +
+				"given to the ProcessManager constructor!"
+			);
+		} else if (!('maxProcesses' in options) || !('isChatBased' in options)) {
+			Monitor.debug(
+				"An options object given to the ProcessManager constructor is " +
+				"missing required properties! The filename given is: " + (options.execFile || '""') + "."
+			);
+		}
 
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = options.execFile;
+		this.maxProcesses = options.maxProcesses || 0;
```

Comment:
Yeah, like that funny looking sideways one next to it. You mean `options.maxProcesses && 0` in case the former's undefined, right?